source/extensions/common/wasm/context.h

```diff
@@ -374,6 +374,19 @@ class Context : public proxy_wasm::ContextBase,
   Http::HeaderMap* getMap(WasmHeaderMapType type);
   const Http::HeaderMap* getConstMap(WasmHeaderMapType type);
 
+  void onHeadersModified(WasmHeaderMapType type) {
+    if (type == WasmHeaderMapType::RequestHeaders) {
+      if (!no_automatic_route_refresh_) {
+        clearRouteCache();
+      } else {
+        ENVOY_LOG_FIRST_N(critical, 20,
```

Comment:
This would be very noisy IMO since it's triggered by every callback. Let's not log by default.

Reply:
Yeah. the idea here is to notice the exist plugin developers that there is a behavior change. And only 20 lines logs will be outputed.

But I am fine to remove this.

Reply:
I am thinking should we keep the log to let users to notice the behavior change. Only 20 lines log will be outputed and others will be ignored.

Or we have no way to let the users to notice this. 🤔

Reply:
But there's no behavior changes for most plugins? It only changes if the route actually changed, which is not what you're checking.